esphome/components/emporia_vue/emporia_vue.cpp

```diff
@@ -8,114 +9,131 @@ namespace emporia_vue {
-
-  // TODO: Log phases
-  for (PhaseConfig *phase : this->phases_) {
-    //LOG_SENSOR("  ", "Phase", phase);
+  ESP_LOGCONFIG(TAG, "  Sensor Poll Interval: %dms", this->sensor_poll_interval_);
+
+  for (auto *phase : this->phases_) {
+    std::string wire;
+    switch (phase->get_input_wire()) {
+      case PhaseInputWire::BLACK:
+        wire = "BLACK";
+        break;
+      case PhaseInputWire::RED:
+        wire = "RED";
+        break;
+      case PhaseInputWire::BLUE:
+        wire = "BLUE";
+        break;
+    }
+    ESP_LOGCONFIG(TAG, "  Phase Config");
+    ESP_LOGCONFIG(TAG, "    Wire: %s", wire.c_str());
+    ESP_LOGCONFIG(TAG, "    Calibration: %f", phase->get_calibration());
+    LOG_SENSOR("    ", "Voltage", phase->get_voltage_sensor());
   }
 
-
-  for (CTSensor *ct_sensor : this->ct_sensors_) {
+  for (auto *ct_sensor : this->ct_sensors_) {
     LOG_SENSOR("  ", "CT", ct_sensor);
-    // TODO: Log other details
+    ESP_LOGCONFIG(TAG, "    Phase Calibration: %f", ct_sensor->get_phase()->get_calibration());
+    ESP_LOGCONFIG(TAG, "    CT Port Index: %d", ct_sensor->get_input_port());
   }
 }
 
 void EmporiaVueComponent::setup() {
   global_emporia_vue_component = this;
 
-  this->i2c_data_queue_ = xQueueCreate(1, sizeof(EmporiaSensorData));
-  xTaskCreatePinnedToCore(&EmporiaVueComponent::i2c_request_task,
-                          "i2c_request_task",
-                          8192,
-                          nullptr,
-                          0,
-                          nullptr,
-                          1
-  );
+  this->i2c_data_queue_ = xQueueCreate(1, sizeof(SensorReading));
+  xTaskCreatePinnedToCore(&EmporiaVueComponent::i2c_request_task, "i2c_request_task", 4096, nullptr, 0, nullptr, 1);
 }
 
 void EmporiaVueComponent::i2c_request_task(void *pv) {
-  const uint32_t NULL_CHECKSUM = 4294967295;
-  TickType_t xLastWakeTime;
-  const TickType_t xDelay = 240 / portTICK_PERIOD_MS;
-  uint32_t last_checksum = NULL_CHECKSUM;
+  const TickType_t poll_interval = global_emporia_vue_component->get_sensor_poll_interval() / portTICK_PERIOD_MS;
+  TickType_t last_poll;
+  uint8_t last_sequence_num = 0;
 
   while (true) {
-    xLastWakeTime = xTaskGetTickCount();
-    EmporiaSensorData data;
-    i2c::ErrorCode error = global_emporia_vue_component->read(reinterpret_cast<uint8_t *>(&data), sizeof(data));
-
-    if (data.read_flag == 0) {
-      last_checksum = NULL_CHECKSUM;
-    }
-
-    // Discard messages not ending by 0x0000
-    if (error == i2c::ErrorCode::ERROR_OK && data.read_flag == 3 && (last_checksum != data.checksum) && data.end == 0) {
-      last_checksum = data.checksum;
-      xQueueOverwrite(global_emporia_vue_component->i2c_data_queue_, &data);
+    last_poll = xTaskGetTickCount();
+    SensorReading sensor_reading;
+
+    i2c::ErrorCode err =
+        global_emporia_vue_component->read(reinterpret_cast<uint8_t *>(&sensor_reading), sizeof(sensor_reading));
+
+    if (err != i2c::ErrorCode::ERROR_OK) {
```

Comment:
This behaves pretty much the same way, except it replaces the checksum for the sequence number. I wasn't really sure how the checksum was supposed to work, do you think we should keep using it?

Reply:
No, I don't think that it's actually a checksum - I'm trying to check if I can correlate any of those bits with a poll rate. I've noticed with my Logic Analyser that the Vue2 doesn't actually poll at a constant rate, and sometimes it pulls after 220ms or 230ms.